File: src/main/java/net/onelitefeather/stardust/command/commands/GodmodeCommand.java
package net.onelitefeather.stardust.command.commands;

import net.kyori.adventure.text.Component;
import net.onelitefeather.stardust.StardustPlugin;
import net.onelitefeather.stardust.util.Constants;
import net.onelitefeather.stardust.util.PlayerUtil;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.incendo.cloud.annotation.specifier.Greedy;
import org.incendo.cloud.annotations.Argument;
import org.incendo.cloud.annotations.Command;
import org.incendo.cloud.annotations.CommandDescription;
import org.incendo.cloud.annotations.Permission;

public class GodmodeCommand {

    private final StardustPlugin plugin;

    public GodmodeCommand(StardustPlugin plugin) {
        this.plugin = plugin;
    }

    @Command("godmode [player]")
    @Permission("stardust.command.godmode")
    @CommandDescription("Makes a player invulnerable to everything")
    public void handleCommand(CommandSender commandSender, @Greedy @Argument(value = "player") Player target) {
        Player actualTarget = target != null ? target : (Player) commandSender;
        handleInvulnerability(commandSender, actualTarget);
    }

    private void handleInvulnerability(CommandSender commandSender, Player target) {
        if (!commandSender.equals(target) && !commandSender.hasPermission("stardust.command.godmode.others")) {
            commandSender.sendMessage(
                    Component.translatable("plugin.not-enough-permissions")
                            .arguments(plugin.getPrefix())
            );
            return;
        }

        target.setInvulnerable(!target.isInvulnerable());
        PlayerUtil.removeEnemies(target, Constants.RADIUS_REMOVE_ENEMIES);

        Component targetEnabledMessage = Component.translatable("commands.godmode.enable.target")
                .arguments(plugin.getPrefix());
        Component targetDisabledMessage = Component.translatable("commands.godmode.disable.target")
                .arguments(plugin.getPrefix());

        Component enabledMessage = Component.translatable("commands.godmode.enable")
                .arguments(plugin.getPrefix(), target.displayName());
        Component disabledMessage = Component.translatable("commands.godmode.disable")
                .arguments(plugin.getPrefix(), target.displayName());

        if (commandSender.equals(target)) {
            target.sendMessage(target.isInvulnerable() ? targetEnabledMessage : targetDisabledMessage);
        } else {
            commandSender.sendMessage(target.isInvulnerable() ? enabledMessage : disabledMessage);
            target.sendMessage(target.isInvulnerable() ? targetEnabledMessage : targetDisabledMessage);
        }
    }
}

File: src/main/kotlin/net/onelitefeather/stardust/command/commands/GodmodeCommand.kt
package net.onelitefeather.stardust.command.commands

import net.kyori.adventure.text.Component
import net.onelitefeather.stardust.StardustPlugin
import net.onelitefeather.stardust.util.PlayerUtils
import net.onelitefeather.stardust.util.RADIUS_REMOVE_ENEMIES
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.incendo.cloud.annotation.specifier.Greedy
import org.incendo.cloud.annotations.Argument
import org.incendo.cloud.annotations.Command
import org.incendo.cloud.annotations.CommandDescription
import org.incendo.cloud.annotations.Permission

class GodmodeCommand(private val stardustPlugin: StardustPlugin) : PlayerUtils {

    @Command("godmode [player]")
    @Permission("stardust.command.godmode")
    @CommandDescription("Makes a player invulnerable to everything")
    fun handleCommand(commandSender: CommandSender, @Greedy @Argument(value = "player") target: Player?) {
        handleInvulnerability(commandSender, target ?: commandSender as Player)
    }

    private fun handleInvulnerability(commandSender: CommandSender, target: Player) {

        try {
            if (target != commandSender && !commandSender.hasPermission("stardust.command.godmode.others")) {
                commandSender.sendMessage(Component.translatable("plugin.not-enough-permissions").arguments(stardustPlugin.getPluginPrefix()))
                return
            }

            target.isInvulnerable = !target.isInvulnerable
            removeEnemies(target, RADIUS_REMOVE_ENEMIES)

            val targetEnabledMessage = Component.translatable("commands.godmode.enable.target").arguments(stardustPlugin.getPluginPrefix())
            val targetDisabledMessage = Component.translatable("commands.godmode.disable.target").arguments(stardustPlugin.getPluginPrefix())

            val enabledMessage = Component.translatable("commands.godmode.enable").arguments(stardustPlugin.getPluginPrefix(), target.displayName())
            val disabledMessage = Component.translatable("commands.godmode.disable").arguments(stardustPlugin.getPluginPrefix(), target.displayName())

            if(commandSender == target) {
                target.sendMessage(if (target.isInvulnerable) targetEnabledMessage else targetDisabledMessage)
            } else {
                commandSender.sendMessage(if (target.isInvulnerable) enabledMessage else disabledMessage)
                target.sendMessage(if (target.isInvulnerable) targetEnabledMessage else targetDisabledMessage)
            }
        } catch (e: Exception) {
            this.stardustPlugin.logger.throwing(GodmodeCommand::class.java.simpleName, "handleInvulnerability", e)
        }
    }
}